python_modules/dagster/dagster/_core/definitions/definitions_class.py

```diff
@@ -375,6 +379,9 @@ class Definitions(IHaveNew):
             Default loggers for jobs. Individual jobs
             can define their own loggers by setting them explictly.
 
+        metadata (Optional[Mapping[str, MetadataValue]]):
+            Arbitrary metadata to attach to the Definitions object.
```

Comment:
Ok if we are going to do this we should describe how metadata manifests in the product (it doesn't) and why this would be useful.

Reply:
I'm simply following the lead of `RepositoryDefinition" here:

```
metadata (Optional[MetadataMapping]): A map of arbitrary metadata for the repository.
```

Since the only current use for `Definitions.metadata` is the private code location metadata API, maybe best to just be vague here? Or should I explicitly say "not rendered in UI" or something?

Reply:
Yeah the docblock on `RepositoryDefinition` is also bad. "Not displayed in UI but accessible on `RepositoryDefinition` at runtime" or something.